System test

aluu317 opened the pull request.

lbustelo commented on Makefile:
```diff
@@ -152,7 +152,7 @@ ifdef SAUCE_USER_NAME
 	@npm run test-sauce --silent -- --sauce-tunnel-id \"$(TRAVIS_JOB_NUMBER)\" --sauce-username $(SAUCE_USER_NAME) --sauce-access-key $(SAUCE_ACCESS_KEY)
 else
-	@npm run test -- --local chrome
+	$(MAKE) -e system-test-local
```
I think this should not go under this target. This target is meant for running unit tests and wil take place before any dist packaging is done within `make all`. System test should be at the end of the entire process.

lbustelo commented on Makefile:
```diff
@@ -258,21 +258,40 @@ else
+	@echo 'Running system integration tests...'
+	@npm run system-test -- --baseurl $(BASEURL) --server $(TEST_SERVER) || (docker rm -f $(SERVER_NAME); -kill `cat SELENIUM_PID`; rm SELENIUM_PID; exit 1)
+	@echo 'System integration tests complete.'
+	@docker rm -f $(SERVER_NAME); kill `cat SELENIUM_PID`; rm SELENIUM_PID
```
You probably need a line here to kill selenium

aluu317 replied:
Line 276 is to kill selenium in case of tests failing, and line 278 kills selenium in case of tests passing

lbustelo commented on .travis.yml:
```diff
@@ -17,7 +17,7 @@ before_script:
 
 script:
-  - make all
+  - BASEURL=http://127.0.0.1:9500; make -e all
```
To simplify, we can assume that `make all` is a target meant for travis. That way you would not need to change anything.

lbustelo commented:
Add `SELENIUM_PID` to `.gitignore`

Commit:
Add integration tests, run locally. fixes #93
disable urth-viz-cloud test, which was failing intermittently. refs #116
(c) Copyright IBM Corp. 2015

lbustelo commented on Makefile:
```diff
@@ -139,7 +139,7 @@ sdist: dist
 		$(REPO) bash -c '$(SETUP_CMD) cp -r /src /tmp/src && \
 			cd /tmp/src && \
-			python setup.py sdist $(POST_SDIST) && \
+			python setup.py sdist && \
```
Mistake in merge... you need that `$(POST_SDIST)` there

lbustelo commented on Makefile:
```diff
@@ -256,14 +256,36 @@ ifdef SAUCE_USER_NAME
+	@echo 'System integration tests complete.'
+	@docker rm -f $(SERVER_NAME)
+	-@kill `cat SELENIUM_PID`
+	@rm SELENIUM_PID
```
You probably want a `-@rm SELENIUM_PID` to avoid failing the build due to missing PID. Maybe impossible to happen, but not a bad idea.

lbustelo commented on Makefile:
```diff
@@ -256,14 +256,36 @@ ifdef SAUCE_USER_NAME
+	@echo 'Waiting 20 seconds for server to start...'
+	@sleep 20
+	@echo 'Running system integration tests...'
+	@npm run system-test -- --baseurl $(BASEURL) --server $(TEST_SERVER) || (docker rm -f $(SERVER_NAME) && kill `cat SELENIUM_PID` && rm SELENIUM_PID && exit 1)
```
Seems redundant... but we can fix later

peller replied:
agreed

lbustelo commented on system-test/system-test-specs.js:
```diff
@@ -89,17 +77,107 @@ describe('system-test (' + desired.browserName + ')', function() {
-        if (result.sauceJobStatus) {
-            result = result.sauceJobStatus(allPassed);
-        }
+        // if (result.sauceJobStatus) {
```
Remove this commented code

peller replied:
@aluu317 why was this commended out?  was this causing the second test to fail?  if so, are we depending on these tests running in a certain order?

lbustelo commented:
So several problems...

1. We got a unit test that is failing sporacally... I say take it offline if it is not deterministic and open an issue to fix it.

<img width="691" alt="screen shot 2015-11-20 at 1 56 03 pm" src="https://cloud.githubusercontent.com/assets/2474841/11310324/92ac4b70-8f8e-11e5-8978-8bae52709da3.png">

2. The System test seem to hang for a very long time on a page that does not load...

<img width="1089" alt="screen shot 2015-11-20 at 1 51 31 pm" src="https://cloud.githubusercontent.com/assets/2474841/11310339/a511b55c-8f8e-11e5-9112-33538d2d0f05.png">

I can confirm that the server is running...
<img width="1010" alt="screen shot 2015-11-20 at 1 52 33 pm" src="https://cloud.githubusercontent.com/assets/2474841/11310347/b3b929d2-8f8e-11e5-8b76-1a7001270a5d.png">

peller commented:
Committed @lbustelo's patch to wait for the notebook using shell script

lbustelo commented:
LGTM!
